Added scanTypeNames() using FastClasspathScanner Library

abibell opened the pull request:
JaversBuilder now supports scanTypeNames() using FastClasspathScanner Library. Default implementations of JaversMongoAutoConfiguration and JaversSqlAutoConfiguration support package scanning using `packagesToScan` configuration property. `packagesToScan` supports multiple values separated by comma

Resolving https://github.com/javers/javers/issues/263

bartoszwalacik commented on javers-spring-boot-starter-sql/src/test/java/org/javers/spring/boot/sql/TestApplication.java:
```diff
@@ -1,15 +1,14 @@
 package org.javers.spring.boot.sql;
 
-import com.google.common.collect.Maps;
```
please don't commit unnecessary changes, it makes code review harder

abibell replied:
Yup removing some of warnings in code is unrelated

bartoszwalacik commented on javers-spring-boot-starter-sql/src/test/java/org/javers/spring/boot/sql/TestApplication.java:
```diff
@@ -22,7 +21,7 @@ public CommitPropertiesProvider commitPropertiesProvider() {
             @Override
             public Map<String, String> provide() {
-                Map props = new HashMap();
+                Map<String, String> props = new HashMap<String, String>();
```
please don't commit changes unrelated to main issue

abibell replied:
Yup removing some of warnings in code is unrelated

bartoszwalacik commented on javers-core/src/main/java/org/javers/core/JaversBuilder.java:
```diff
@@ -233,22 +233,19 @@ public JaversBuilder registerValueObject(ValueObjectDefinition valueObjectDefini
+    	if(packagesToScan==null || packagesToScan.trim().isEmpty())
+    		return this;
+		
+		Class<?>[] list = ReflectionUtil.getClasses(TypeName.class, packagesToScan.split(","));
```
as you can see, indentation is messed up (use spaces, not tabs)

bartoszwalacik commented on javers-core/src/main/java/org/javers/core/JaversBuilder.java:
```diff
@@ -233,22 +233,19 @@ public JaversBuilder registerValueObject(ValueObjectDefinition valueObjectDefini
-                "If you think that this method should be implemented, " +
-                "vote here: https://github.com/javers/javers/issues/263");
+    public JaversBuilder scanTypeNames(String packagesToScan){
+    	if(packagesToScan==null || packagesToScan.trim().isEmpty())
```
```
if (...) {
    return ...
}
```

bartoszwalacik commented on javers-core/src/main/java/org/javers/common/reflection/ReflectionUtil.java:
```diff
@@ -174,6 +184,21 @@ private static boolean isProtected(Member member){
+			}
+    		classes.add(classType);
+    	}
+    	return null;
```
this method (clue of your PR) always returns null.
Write tests, check if they fail, and then implement a new feature.

bartoszwalacik commented:
we will continue work on abibell-master branch (https://github.com/javers/javers/tree/abibell-master). I'm going to push some tests

abibell commented:
I addressed some feedback. I realised that my Eclipse didn't recognize groovy files. Anyway its all yours. Revert commits, do whatever you want on it.

bartoszwalacik commented:
merged from another branch https://github.com/javers/javers/pull/434
this feature will be released in JaVers 2.3 (tomorrow)
